### PR TITLE
feat: add no implicit dependencies

### PR DESCRIPTION
**Pull request by XVincentX**
…so we do not require dev dependencies in production code

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -29,17 +30,29 @@
...
-    "max-classes-per-file": [true, 3],
+    "max-classes-per-file": [
+      true,
+      3
+    ],
     "no-var-requires": false,
     "member-ordering": false,
     "no-floating-promises": true,
-    "interface-over-type-literal": false
+    "no-implicit-dependencies": [
```
Comment (marbemac) [inline tslint.json:40]: Could you also add `__stories__` to this list?

File: tslint.json [hunk 2]
```diff
@@ -29,17 +30,29 @@
...
     "ban-types": false,
-    "max-classes-per-file": [true, 3],
+    "max-classes-per-file": [
+      true,
+      3
+    ],
     "no-var-requires": false,
     "member-ordering": false,
     "no-floating-promises": true,
-    "interface-over-type-literal": false
```
Comment (marbemac) [inline tslint.json:36]: Did you mean to remove this?

**Review by marbemac — CHANGES_REQUESTED** [review]

**Comment by XVincentX** [comment]
@marbemac Gosh, I was just trying this locally, there's a big problem. Lodash, for example, we include their typings as a devDependency because the package itself does not ship with any of these.

This results in an error from tslint — because we're using a `devDepenendecy` 🤔 

Got any idea?

File: tslint.json (hunk 2 above)
Comment (XVincentX) [reply tslint.json:36]: It's false by default, right?

**Comment by XVincentX** [comment]
https://github.com/palantir/tslint/issues/3375 :(

**Comment by marbemac** [comment]
Ah OK then I suggest we pass on this for right now - that will indeed be a big headache. We have a wishlist item to investigate using eslint for linting (now that typescript team recommends it). Suggest we wait until we have time for that larger investigation?

**Comment by XVincentX** [comment]
Yeah that makes sense. I really wanted to have this inside thought :(

**Pull request closed without merging**